src/aws_signature.erl

```diff
@@ -263,7 +265,13 @@ sign_v4_query_params(AccessKeyID, SecretAccessKey, Region, Service, DateTime, Me
         end,
     BaseParams =
         [{<<"X-Amz-Algorithm">>, <<"AWS4-HMAC-SHA256">>},
-         {<<"X-Amz-SignedHeaders">>, <<"host">>}],
+         {<<"X-Amz-SignedHeaders">>,
+          if Tag == undefined ->
+                  <<"host">>;
+             Tag =/= undefined ->
+                  hackney_url:urlencode(<<"host;x-amz-tagging">>)
```

Comment:
Are you sure this works? I'd expect something like below instead:
```erlang
BaseParams0 =
        [{<<"X-Amz-Algorithm">>, <<"AWS4-HMAC-SHA256">>},
         {<<"X-Amz-SignedHeaders">>, <<"host">>}],
    BaseParams = maybe_add_tag_header(BaseParams0, Tag),
```

With the `maybe_add_tag_header/2` being defined as something like below (leaving aside the `hackney_url` part but I'd expect the header to be a part of this? 
```erlang
maybe_add_tag_header(Headers, undefined) ->
    Headers;  
maybe_add_tag_header(Headers, Tag) ->
    [{<<"x-amz-tagging">>, hackney_url:urlencode(<<"host;x-amz-tagging">>} | Headers].
```

Reply:
That wouldn't be enough since we need to add X-Amz-Tagging to X-Amz-SignedHeaders as defined here: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html

Reply:
Yes, you're right; it should be a list of tags.